cscs-checks/system/basic/file_system_check.py

```diff
@@ -0,0 +1,182 @@
+        self.directory_name = osext.expandvars('$' + self.directory)
+        self.perf_patterns = {
+            'real_time': sn.extractsingle(r'real (?P<real_time>\S+)',
+                                          self.stderr, 'real_time', float),
+            'size': sn.extractsingle(
+                r'(?P<size>\S+).+'+re.escape(self.directory_name),
+                self.stdout, 'size', float)
+        }
+
+        # TODO: system is not always relevant
+        self.reference = {
+            'PROJECT': {
+                'size': (1000, None, 0.1, 'MB'),
+                'real_time': (5.0, None, 0.1, 's')
+            },
+            'HOME': {
+                'size': (900, None, 0.1, 'MB'),
+                'real_time': (5.0, None, 0.1, 's')
+            },
+            'SCRATCH': {
+                'size': (900, None, 0.1, 'MB'),
+                'real_time': (5.0, None, 0.1, 's')
+            }
+        }
+        self.executable_opts = ['/usr/bin/du -mhs --block-size=1M',
+                                self.directory_name]
+
+    @rfm.run_before('sanity')
+    def set_sanity(self):
+        self.sanity_patterns = sn.assert_found(self.directory_name,
+                                               self.stdout)
+
+
+@rfm.simple_test
+class fs_check_touch_file(FileSystemCommandCheck):
+    directory = parameter(['SCRATCH'])
+
+    def __init__(self):
+        super().__init__()
+        self.test_file = osext.expandvars('$' + self.directory +
+                                          '/reframe_touch_test_file')
+        self.executable_opts = ['touch', self.test_file]
+
+    @rfm.run_after('run')
+    def delete_test_file(self):
+        if os.path.exists(self.test_file):
+            os.remove(self.test_file)
+
+
+@rfm.simple_test
+class fs_check_cat_file(FileSystemCommandCheck):
+    file = parameter(['/apps/daint/system/etc/BatchDisabled',
+                      '/etc/opt/slurm/cgroup.conf',
+                      '/etc/opt/slurm/plugstack.conf',
+                      '/etc/opt/slurm/slurm.conf',
+                      '/etc/opt/slurm/topology.conf',
+                      '/etc/opt/slurm/node_prolog.sh',
+                      '/etc/opt/slurm/node_epilog.sh',
+                      '/etc/opt/slurm/gres.conf'])
+
+    # TODO: find correct test name
+    def __init__(self):
+        super().__init__()
+        self.reference = {
+            'daint:login': {
+                # TODO: real times have large variances,
+                # do we need a specific refererence for each test?
+                'real_time': (0.05, None, 0.1, 's')
+            },
+            'dom:login': {
+                'real_time': (0.05, None, 0.1, 's')
+            }
+        }
+        self.executable_opts = ['cat', self.file, ' > /dev/null']
+
+
+# TODO: this test is almost identical to the cat one
+@rfm.simple_test
+class fs_check_find_dir(FileSystemCommandCheck):
+    directory = parameter([
+        '/project/csstaff',
```

Comment:
I think we need to choose another path for the Jenkins user.
```bash
> find /project/csstaff/
/project/csstaff/
find: ‘/project/csstaff/’: Permission denied
```

Reply:
Do you think that /project is a relevant path in this case?